controller/controller_server/csi_controller_server.py

```diff
@@ -136,32 +123,9 @@ def CreateVolume(self, request, context):  # pylint: disable=too-many-branches
                 res = utils.generate_csi_create_volume_response(volume, source_type)
                 logger.info("finished create volume")
                 return res
-
-        except (controller_errors.IllegalObjectName, controller_errors.SpaceEfficiencyNotSupported,
-                controller_errors.PoolDoesNotExist, controller_errors.PoolDoesNotMatchCapabilities,
-                controller_errors.PoolParameterIsMissing, controller_errors.ExpectedSnapshotButFoundVolumeError) as ex:
-            logger.exception(ex)
-            context.set_details(ex.message)
-            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
-            return csi_pb2.CreateVolumeResponse()
-        except controller_errors.ObjectNotFoundError as ex:
-            context.set_code(grpc.StatusCode.NOT_FOUND)
-            context.set_details(ex.message)
-            return csi_pb2.CreateVolumeResponse()
-        except controller_errors.PermissionDeniedError as ex:
-            context.set_code(grpc.StatusCode.PERMISSION_DENIED)
-            context.set_details(ex.message)
-            return csi_pb2.CreateVolumeResponse()
-        except controller_errors.VolumeAlreadyExists as ex:
-            context.set_details(ex.message)
-            context.set_code(grpc.StatusCode.ALREADY_EXISTS)
-            return csi_pb2.CreateVolumeResponse()
-        except controller_errors.NotEnoughSpaceInPool as ex:
-            logger.exception(ex)
-            context.set_details(ex.message)
-            context.set_code(grpc.StatusCode.RESOURCE_EXHAUSTED)
-            return csi_pb2.CreateVolumeResponse()
         except Exception as ex:
+            if type(ex) in exception_handler.status_codes_by_exception.keys():
+                raise
             logger.error("an internal exception occurred")
             logger.exception(ex)
             context.set_code(grpc.StatusCode.INTERNAL)
```

Comment:
I think this should be handled by the decorator

Reply:
This isn't a common exception.

Reply:
if by `This` you mean the `Exception` type, then inheritance-wise it is very common

Reply:
Done